### PR TITLE
[ci][api] Disallow user password change in LDAP mode.

### PR DESCRIPTION
**Pull request by evanrolfe**
https://trello.com/c/ljYDOGC7/478-3-p1-drop-editing-ldap-user-accounts

### CODE REVIEW

**Comment by mdeniz** [comment]
This commit should have the [api] tag also...

**Comment by bgeuken** [comment]
The wording of the commit message sounds strange to me. Is there a mistake or is it me?

**Comment by evanrolfe** [comment]
> The wording of the commit message sounds strange to me. Is there a mistake or is it me?

I'm limited to 72 characters for the commit subject thanks to our new friend git-cop :/ I really don't think I can express what i want to say in the subject in less than 72 chars..

**Comment by bgeuken** [comment]
I see:/

**Comment by bgeuken** [comment]
Maybe 'Permit user creation in LDAP mode' and some more detailed explanation in the commit message?

File: src/api/app/controllers/person_controller.rb [hunk 1]
```diff
@@ -155,6 +155,15 @@ class NoPermission < APIException
   end
 
   def internal_register
+    if CONFIG['ldap_mode'] == :on
+      render_error(
+        status: 403,
+        errorcode: 'permission_denied',
+        message: "User accounts can not be registered via OBS when in LDAP mode. Please refer to your LDAP server to change it."
```
Comment (bgeuken) [inline src/api/app/controllers/person_controller.rb]: How about "... . Please refer to your LDAP server to create new users." or " to register new users"?

File: src/api/app/controllers/person_controller.rb [hunk 2]
```diff
@@ -155,6 +155,15 @@ class NoPermission < APIException
   end
 
   def internal_register
+    if CONFIG['ldap_mode'] == :on
```
Comment (mschnitzer) [inline src/api/app/controllers/person_controller.rb]: Better use `::Configuration.ldap_enabled?`

File: src/api/app/controllers/person_controller.rb (hunk 1 above)
Comment (evanrolfe) [inline src/api/app/controllers/person_controller.rb]: Changed.

File: src/api/app/controllers/person_controller.rb (hunk 2 above)
Comment (evanrolfe) [inline src/api/app/controllers/person_controller.rb]: Changed.

**Comment by evanrolfe** [comment]
[(3) P1: Drop editing LDAP user accounts](https://trello.com/c/ljYDOGC7/478-3-p1-drop-editing-ldap-user-accounts)

**Pull request merged**